src/test/run-make/crate-hash-rustc-version/Makefile

```diff
@@ -1,23 +1,28 @@
 -include ../../run-make-fulldeps/tools.mk
 
-# only-linux
-
-# This test only works on Linux, since it uses lib*.so files.
-
 FLAGS := -Cprefer-dynamic -Zsymbol-mangling-version=v0
+UNAME := $(shell uname)
+ifeq ($(UNAME),Linux)
+  EXT=".so"
+  NM_CMD="nm -d"
```

Comment:
This needs to be capital `-D`. Also I don't think `nm` is always available.

Reply:
Should we remove the `nm` logic entirely?

Reply:
Or do `$(NM_CMD) $(TMPDIR)/liba$(EXT) > $(TMPDIR)/symbolsbefore || true`?